### ISSUE TITLE
Modalize Edit Company, Edit Review, Edit Application

### ISSUE THREAD

**Issue by theandrewchan** [codebase-berkeley/CodeDoor#70]
Editing a company/review/application should be modalized to use the new company and review modal components.

This will involve updating the `edit_company`/`edit_review`/`edit_application` view functions to return a JSON of success/failure instead of a redirect on POST, and should return a 404 instead of the edit form on GET. Additionally we will have to add more parameters to the template components to allow prefilling of the modal inputs (alternatively we could prefill using JS and an extra call to the backend...).

Note that some of these pages don't even work right now - the edit review and edit application pages for example are impossible to use because they still use the old company dropdown (which literally runs `SELECT * from companies` to prefill a dropdown) instead of the new one where you search for a company and select from a list of matching companies.

**Comment by theandrewchan** [comment]
There's also an edit question page, but I'm considering completely removing question functionality at this time, so changing that is not a priority,

**Comment by theandrewchan** [comment]
Fixed with https://github.com/codebase-berkeley/CodeDoor/commit/21bb650df291122f11e7b3f15c0081ca89451e81

Notes:
- I've removed frontend question functionality, so questions will no longer be displayed for each review and there isn't the option to create a question anymore. The question views and URLs still exist on the backend.
- Since the edit company page works without modification right now, I won't be modalizing it yet. This will be more worth changing if/when we move to a more permanent frontend framework in the future.
